internal/jimm/jujuauth/migrationtokens.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Canonical.
+
+package jujuauth
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/names/v5"
+
+	"github.com/canonical/jimm/v3/internal/jimmjwx"
+)
+
+const (
+	migrationExpiry = 3 * time.Hour
+)
+
+// MigrationTokenGenerator is responsible for generating migration tokens.
+// It uses the JWTService to create a token with specific claims for migration purposes.
+// The token includes the user, model tag, JIMM's UUID and an indication that it is for migration.
+type MigrationTokenGenerator struct {
```

Comment:
Calling the interface and the implementation the same name is kinda confusing. Maybe call this MigrationTokenService or something idk.